src/main/java/com/google/enterprise/secmgr/authncontroller/AuthnSessionManagerImpl.java

```diff
@@ -39,80 +47,77 @@
 public final class AuthnSessionManagerImpl implements AuthnSessionManager {
   private static final Logger logger = Logger.getLogger(AuthnSessionManagerImpl.class.getName());
 
-  // User to session id map.
-  @GuardedBy("this") private final Map<AuthnPrincipal, String> userSessionMap;
-  @GuardedBy("this") private final Map<String, AuthnSession> sessionMap;
-  @GuardedBy("this") private final Map<AuthnSession, Long> refTimeMap;
   @GuardedBy("this") private long sessionIdleMillis;
 
+  private static final Random random = new SecureRandom();
+
+  @Inject
+  private RedisRepository redisRepository;
```

Comment:
I think this cl may be broken into two cls:
 - Change which party creates sessionid
 - Refactor some saml code, Introduce redis as storage option, serialization efficiencies. Refactoring is made only as a part of redis storage introduction. "Serialization efficiencies" - it's not optional, without that change I could not serialize session because SAML objects do not implement Serialazible interface.

Reply:
If the SAML logic refactor is sensible, then it seems that it can happen in isolation.  Why not have the SAML refactor simply be a seperate CL?

Reply:
Does secmgr, before your changes, have a way to write a session to XML or to JSON?